src/ml/neural_net/image_augmentation.hpp

```diff
@@ -48,10 +48,17 @@ struct image_box {
   // area of the other image_box.
   void extend(const image_box& other);
 
+  // Set image dimentsion
+  void set_image_dimension(float image_width, float image_height);
+
   float x = 0.f;
   float y = 0.f;
   float width = 0.f;
   float height = 0.f;
+
+  // Store true image width and height
```

Comment:
I don’t think we should store this state here. This type is meant to be closer to `CGRect` conceptually. See other comments below for other approaches to this bug

Reply:
good!